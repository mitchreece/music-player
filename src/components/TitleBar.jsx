import React from 'react'
import { connect } from 'react-redux'

function mapStateToProps(state) {
	return { 
		title: state.title,
	} 
}

const TitleBar = (props) => {
	return (
		<header className="title-bar">
			<p className="title-bar__title">{props.title}</p>
		</header>
	)
}

export default connect(mapStateToProps)(TitleBar)
