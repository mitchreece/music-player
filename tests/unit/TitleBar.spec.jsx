import React from 'react'
import configureStore from 'redux-mock-store'
import renderer from 'react-test-renderer'
import TitleBar from '../../src/components/TitleBar'

const middlewares = []
const mockStore = configureStore(middlewares)

const initialState = {
	title: 'Google Play Music Desktop App',
}

const store = mockStore(initialState)

it('renders correctly', () => {
	const tree = renderer
		.create(<TitleBar store={store} />)
		.toJSON()

	expect(tree).toMatchSnapshot()
})
