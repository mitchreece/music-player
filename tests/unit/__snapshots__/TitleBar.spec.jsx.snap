// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<header
  className="title-bar"
>
  <p
    className="title-bar__title"
  >
    Google Play Music Desktop App
  </p>
</header>
`;
